### DISCUSSION TITLE
Can the static build be decompiled?

### DISCUSSION THREAD

**Discussion by sawirricardo** [php/frankenphp#705] [Q&A] [answered]
I am discussing this with my friends and talking about how we can ship our apps to server we do not own. We don't want to allow the buyers of our app break open the app, let alone steal the app code.

One idea is to use the frankenphp's static builder, which can compile this to single binary file. However, can this prevent users to not "decompile" the app?

Thank you

**Comment by withinboredom** [comment, marked as answer]
The PHP is decompressed and stored in temporary files. They can read the source code. There are obfuscation tools for php. In general though, it is better to rely on contracts for this sort of thing instead of technical hurdles.

**Comment by razvanphp** [comment]
I've read this in the docs: `... compress the resulting binary using UPX`. Isn't this supposed to help with obfuscation? Can we prevent the decompression part?

**Reply by withinboredom** [reply]
There's no obfuscation here. If you need obfuscation, you need to use a dedicated PHP obfuscation tool.
